src/expr/src/relation/func.rs

```diff
@@ -962,12 +962,12 @@ fn generate_series_ts(
     step: Interval,
     conv: fn(NaiveDateTime) -> Datum<'static>,
 ) -> Result<impl Iterator<Item = (Row, Diff)>, EvalError> {
-    if step.months == 0 && step.duration == 0 {
+    if step.months == 0 && step.days == 0 && step.micros == 0 {
```

Comment:
These need to include days. Probably should combine days and times here, check PG.